doc/connectivity/bluetooth/bluetooth-contrib.rst

```diff
@@ -0,0 +1,55 @@
+.. _bluetooth-contrib:
+
+Contributing to the Bluetooth subsystem
+#######################################
+
+Overview
+********
+
+This page describes how one would contribute to the Zephyr Bluetooth stack.
+It is a work in progress.
+
+The first step is familiarizing yourself with the :ref:`Stack architecture <bluetooth-arch>`.
+
+.. _bluetooth-coding-style:
+
+Coding guidelines
+*****************
+
+The following describes a few of the coding conventions we use in the Bluetooth
+subsystem. See also the :ref:`Zephyr Coding Guidelines <coding_guidelines>` and
+the :ref:`Zephyr Coding Style <coding_style>`.
+
+The guidelines in this document should not take precedence on rules from the
+Zephyr Coding Guidelines and Coding Style mentioned above.
+
+Unless there is a convention listed here that disagrees, the general "rule" is
+to try to follow the style of the code surrounding the change.
+
+The Bluetooth contribution guidelines are not set in stone, and this is a living
+document. It will be updated if enough people agree on amending or changing the
+rules.
+
+Tools
+=====
+
+Install and use editorconfig_. This should ensure your editor uses the proper
+settings, e.g. indentation, maximum line length, etc..
+
+Before submitting a pull request, it is a good idea to:
+
+- Run ``clang-format -i`` on your patch. See also the VScode_ or SublimeText_ plugins.
+- Run ``./scripts/ci/check_compliance.py -c upstream/main..**`` and resolve any reported issues
+
+Git `Commit hooks`_ can be used for running the compliance script automatically.
+
+Whitespace
+==========
+
+- Tabs in the beginning of a line is the only place where tabs should ever be found.
+- Add an empty line after every variable definition block.
```

Comment:
Isn't this a Zephyr-wide requirement?

Reply:
can't find it. where do you see it?

Reply:
I don't. It's just that I remember seeing CI failures when there wasn't an empty line between variable declarations and code.

Reply:
Yes I think that rule is encoded in checkpatch, but apparently not in the zephyr or Linux guidelines.

Reply:
That's true. Will you make the PR to the general guidelines then?

Reply:
> That's true. Will you make the PR to the general guidelines then?

I wasn't planing to, no. Isn't it just a matter of moving that line to the right place and a separate commit? Could even be within this PR, I suppose.

Reply:
It's not. It will drag in other people in the PR with their own nitpicks and ideas about where and what and how this one line should be documented.

Reply:
The zephyr coding style already has a list of exceptions to the Linux style.
https://github.com/zephyrproject-rtos/zephyr/blob/2de6274119c9e908784ac7729da6b215924c9cb5/doc/contribute/guidelines.rst?plain=1#L527
That seems like a good place to enumerate this.